### PR TITLE
Alternative implementation for std::filesystem

### PR DESCRIPTION
**Pull request by AaronVanGeffen**
Implements @chfast's suggestion from #381, ported from https://github.com/OpenRCT2/OpenRCT2/pull/10261

Presented as an alternative to #401.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -227,13 +211,7 @@ if (NOT MINGW)
...
 
 target_compile_definitions(${PROJECT} PRIVATE _NO_LOCO_WIN32_=1)
-if (USE_BOOST_FILESYSTEM)
-    target_link_libraries(${PROJECT} ${Boost_FILESYSTEM_LIBRARY} ${Boost_SYSTEM_LIBRARY})
-    target_include_directories(${PROJECT} PRIVATE ${Boost_INCLUDE_DIRS})
-    target_compile_definitions(${PROJECT} PRIVATE BOOST_SYSTEM_NO_DEPRECATED=1 BOOST_FILESYSTEM_NO_DEPRECATED=1 _OPENLOCO_USE_BOOST_FS_=1)
-else()
-	target_link_libraries(${PROJECT} stdc++fs)
-endif()
+target_link_libraries(${PROJECT} stdc++fs)
```
Comment (AaronVanGeffen) [inline CMakeLists.txt]: This needs to be disabled on macOS, e.g.:

```suggestion
if (NOT APPLE)
    target_link_libraries(${PROJECT} stdc++fs)
endif()
```

**Comment by janisozaur** [comment]
I expect boost to be a rather large part of our docker images, when this goes in it would make sense to remove the package from our images.

**Comment by AaronVanGeffen** [comment]
@janisozaur Glad you like it. I could use some help getting MinGW to work for this PR, by the way.

**Comment by AaronVanGeffen** [comment]
I have tested the AppVeyor builds for Windows, and the Travis build for macOS, as well as my own build for Arch Linux. All work fine; it's just MinGW that needs to pass at this point...

**Pull request merged**